dotCMS/src/integration-test/java/com/dotcms/publisher/bundle/business/BundleAPITest.java

```diff
@@ -0,0 +1,205 @@
+        assertTrue(bundlesDeleted.contains(bundleIdAdmin));
+        assertTrue(bundlesDeleted.contains(bundleIdUser));
+        assertFalse(bundlesDeleted.contains(bundleIdAdmin_futureDate));
+
+        assertNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin_futureDate));
+    }
+
+    @Test
+    public void test_deleteBundleAndDependenciesOlderThan_byLimitedUser() throws DotDataException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin = insertPublishingBundle(adminUser.getUserId(),new Date());
+        final String bundleIdUser = insertPublishingBundle(newUser.getUserId(),new Date());
+        final String bundleIdUser_futureDate = insertPublishingBundle(newUser.getUserId(),DateUtil.addDate(new Date(),
+                Calendar.MONTH,1));
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser_futureDate));
+
+        Set<String> bundlesDeleted = bundleAPI.deleteBundleAndDependenciesOlderThan(new Date(),newUser);
+        assertTrue(bundlesDeleted.contains(bundleIdUser));
+        assertFalse(bundlesDeleted.contains(bundleIdAdmin));
+        assertFalse(bundlesDeleted.contains(bundleIdUser_futureDate));
+
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser_futureDate));
+    }
+
+    @Test
+    public void test_deleteAllBundles_byAdmin() throws DotDataException, DotPublisherException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin_success = insertPublishingBundle(adminUser.getUserId(),new Date());
+        insertPublishAuditStatus(Status.SUCCESS,bundleIdAdmin_success);
+        final String bundleIdUser_success = insertPublishingBundle(newUser.getUserId(),new Date());
+        insertPublishAuditStatus(Status.SUCCESS,bundleIdUser_success);
+        final String bundleIdAdmin_failedPublish = insertPublishingBundle(adminUser.getUserId(),new Date());
+        insertPublishAuditStatus(Status.FAILED_TO_PUBLISH,bundleIdAdmin_failedPublish);
+        final String bundleIdUser_failedBundle = insertPublishingBundle(newUser.getUserId(),new Date());
+        insertPublishAuditStatus(Status.FAILED_TO_BUNDLE,bundleIdUser_failedBundle);
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin_success));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser_success));
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin_failedPublish));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser_failedBundle));
+
+        Set<String> bundlesDeleted = bundleAPI.deleteAllBundles(adminUser,Status.SUCCESS,Status.FAILED_TO_PUBLISH);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'bundlesDeleted' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)